docs/website/docs/developers/debugging/gpu.md

```diff
@@ -0,0 +1,234 @@
+
+## Pinpointing runtime issues
+
+On the other side, if we suspect that it's a runtime issue, here are some
+useful approachs and tips:
+
+!!! tip "[:material-check-bold:/:material-trending-up:]"
+
+    [Tracy profiling](../performance/profiling-with-tracy.md) is a great way to
+    view how the application runs dynamically.
+    It can help to show problematic GPU API call sequences and performance
+    bottlenecks.
+
+    * It requires adding `-DIREE_ENABLE_RUNTIME_TRACING=ON` during CMake
+      configuration, or use the `IREE_PY_RUNTIME=tracy` environment variable
+      when invoking IREE runtime installed via Python packages.
+
+!!! tip "[:material-check-bold:]"
+
+    GPU validation can sometimes give us hints:
+
+    * [:simple-apple:] Enable validation via `export METAL_DEVICE_WRAPPER_TYPE=1`.
+    * [:simple-vulkan:] Use `--vulkan_validation_layers=true` to `iree-run-module`, or
+    * [:simple-vulkan:] Force enable via environment variables to the Vulkan loader:
+      `export VK_INSTANCE_LAYERS=VK_LAYER_LUNARG_standard_validation`
+      (may additionally need
+      `export VK_LAYER_PATH=$VULKAN_SDK/etc/vulkan/explicit_layer.d` and
+      `export LD_LIBRARY_PATH=$VULKAN_SDK/lib` if Vulkan SDK is not installed
+      to a system path).
+
+!!! tip "[:material-check-bold:]"
+
+    Turning on verbose output can give us more information:
+
+    * When compiling IREE runtime, add
+      `-DCMAKE_C_FLAGS=-DIREE_VM_EXECUTION_TRACING_FORCE_ENABLE=1` in CMake
+      configuration to enable VM op tracing.
+    * [:simple-vulkan:] Use `--vulkan_debug_verbosity=4` to `iree-run-module`.
+    * [:simple-vulkan:] Print all Vulkan APIs calls with detailed arguments:
+      `export VK_INSTANCE_LAYERS=VK_LAYER_LUNARG_api_dump`
+      (may additionally need
+      `export VK_LAYER_PATH=$VULKAN_SDK/etc/vulkan/explicit_layer.d` and
+      `export LD_LIBRARY_PATH=$VULKAN_SDK/lib` if Vulkan SDK is not installed
+      to a system path).
+
+!!! tip "[:material-check-bold:]"
+
+    Try different "debugging modes" provided by HAL drivers:
+
+    * [:simple-nvidia:] Switch `--cuda_use_streams=` between `true` and `false` to
+      `iree-run-module` to see whether the issue comes from the stream/graph
+      command buffer implementation.
+    * [:simple-nvidia:] Switch `--cuda_async_allocations=false` to `iree-run-module` to
+      see if the issue comes from async allocation.
+    * [:simple-apple:] Use `--metal_serial_command_dispatch=true`,
+      `--metal_command_buffer_retain_resources=true`, or
+      `--metal_resource_hazard_tracking=true` to `iree-run-module` to see
+      if any of the above "fixes" the issue.
+      It can help to isolate the pontential problem.
+    * [:simple-vulkan:] Use `--vulkan_robust_buffer_access=true` to `iree-run-module`
+      especially when seeing undeterministic/corrupted contents in buffers and
+      suspecting there are buffer allocation/indexing issues.
```

Comment:
FYI you could use content tabs here https://squidfunk.github.io/mkdocs-material/reference/content-tabs/

I didn't want to suggest that for the rest of the page, since there are sections that explicitly say to compare one backend against another. For cases where a user already chose a target, content tabs give you a convenient way to hide other information. For example, see https://iree.dev/developers/general/developer-tips/#dumping-executable-files 
![image](https://github.com/openxla/iree/assets/4010439/604eb664-be9a-42b7-9fea-e4f7c3549813)

Reply:
Yeah. Not sure how applicable this one is. for now I'm happy with just puting everything together. Once we have more tips for different APIs we can change.